### PR TITLE
allow TYP_STRUCT for GT_RETURN - special case CNS_INT 0 initialization

### PR DESCRIPTION
**Pull request by yowl**
This PR is pretty much just as per the title.  Added a case for `TYP_STRUCT` in `buildReturn` with special handling for 0 initialization.

Slight difference to the other location where we 0 initialize structs is that the size here is taken from the LLVM type not the `ClassLayout->GetSize` as we dont have a `GenTreeBlk`

cc @SingleAccretion

### CODE REVIEW

File: src/coreclr/jit/llvm.cpp [hunk 1]
```diff
@@ -1620,12 +1620,23 @@ void Llvm::buildReturn(GenTree* node)
...
                 failFunctionCompilation();
             }
+            if (node->gtGetOp1()->IsIntegralConst(0))
+            {
+                // special case returning 0 initialized structs
+                Type* structLlvmType = getLlvmTypeForCorInfoType(_sigInfo.retType, _sigInfo.retTypeClass);
+                Value* structAddrValue = _builder.CreateAlloca(structLlvmType, 0U);
+                _builder.CreateMemSet(structAddrValue, _builder.getInt8(0), _builder.getInt32(structLlvmType->getScalarSizeInBits() >> 3), {});
+                _builder.CreateRet(_builder.CreateLoad(structAddrValue));
+
```
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]:
```suggestion
                // Special-case returning zero-initialized structs.
                Type* structLlvmType = getLlvmTypeForCorInfoType(_sigInfo.retType, _sigInfo.retTypeClass);
                Value* structAddrValue = _builder.CreateAlloca(structLlvmType, 0U);
                Value* structSizeValue = _builder.getInt32(structLlvmType->getScalarSizeInBits() / BITS_PER_BYTE);
                _builder.CreateMemSet(structAddrValue, _builder.getInt8(0), structSizeValue, {});
                _builder.CreateRet(_builder.CreateLoad(structAddrValue));
```

**Review by SingleAccretion — APPROVED** [review]
LGTM modulo nit.

File: src/coreclr/jit/llvm.cpp (hunk 1 above)
Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: Isn't that changing a division by 8 (going from bits to bytes) to a division by 4 (`TARGET_POINTER_SIZE` for wasm32).  If the `>> 3` is not great, we can just do `/ 8` to go from bits to bytes?

Comment (SingleAccretion) [reply src/coreclr/jit/llvm.cpp]: Ah, sorry, indeed it should be `/ BITS_PER_BYTE` (I edited the suggestion).

**Comment by yowl** [comment]
oops.  Probably should only do the `memset` for `IsIntegralConst(0)` for `TYP_STRUCT`...

**Comment by yowl** [comment]
Green now, @SingleAccretion when you have a moment, please take a look at the last commit to make sure we are happy with this.  Thanks!

**Review by SingleAccretion — APPROVED** [review]

**Comment by jkotas** [comment]
Thank you!

**Pull request merged**